src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/Utilities/DataFlowExtensions/DataFlowExtensionMethodCaller.cs

```diff
@@ -0,0 +1,33 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Threading.Tasks.Dataflow;
+
+namespace Microsoft.VisualStudio.ProjectSystem.Utilities.DataFlowExtensions
+{
+    /// <summary>
+    /// <see cref="DataFlowExtensionMethodCaller"/> is a caller of a <seealso cref="IDataFlowExtensionWrapper"/>,
+    /// which wraps a call to extension methods defined in DataFlowExtensions. These methods cannot be mocked since
+    /// these are static methods. Hence we are covering the static methods with wrapper which makes the behavior non-static
+    /// which will then let us mock
+    /// </summary>
+    public class DataFlowExtensionMethodCaller
```

Comment:
I'm not a fan of this pattern or the way that is is plugged in. The way this should work is that unit testing uses the same constructor as the product. If you need to make something testable than you "import it". If we can't make the data flow unit testable (I'll find out shortly as I'll run into the same thing), then separate the thing that listens to the data flow into a separate component that calls you back.